AstraEngine.Canvas2D/Color.cs

```diff
@@ -3,6 +3,31 @@ namespace AstraEngine.Canvas2D;
 /// <summary>Represents an RGBA color</summary>
 public record struct Color(byte R, byte G, byte B, byte A)
 {
+    public static Color FromHexString(string hex)
+    {
+        if (hex.StartsWith("#"))
+        {
+            hex = hex.Substring(1);
+        }
+
+        if (hex.Length == 8)
+        {
+            byte r = byte.Parse(hex.Substring(0, 2), System.Globalization.NumberStyles.HexNumber);
+            byte g = byte.Parse(hex.Substring(2, 2), System.Globalization.NumberStyles.HexNumber);
+            byte b = byte.Parse(hex.Substring(4, 2), System.Globalization.NumberStyles.HexNumber);
+            byte a = byte.Parse(hex.Substring(6, 2), System.Globalization.NumberStyles.HexNumber);
+            return new Color(r, g, b, a);
+        }
+        else if (hex.Length == 6)
+        {
+            byte r = byte.Parse(hex.Substring(0, 2), System.Globalization.NumberStyles.HexNumber);
+            byte g = byte.Parse(hex.Substring(2, 2), System.Globalization.NumberStyles.HexNumber);
+            byte b = byte.Parse(hex.Substring(4, 2), System.Globalization.NumberStyles.HexNumber);
+            return new Color(r, g, b, 1);
+        }
+        return new Color(1, 0, 1, 1);

```

Comment:
This looks great. However, I would recommend throwing an exception here rather than returning a default color.